src/pages/DealsPage.js

```diff
@@ -1,7 +1,55 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import ItemList from '../components/ItemList';
+import styled from 'styled-components';
+import ContentWrapper from '../styles/contentWrapper';
+import axios from 'axios';
+
+import mockData from '../db/mock_data.json';
+
+const API_URL =
+  'https://gp-super-store-api.herokuapp.com/item/list?sortDir=asc&isOnSale=true';
+
+const StyledContainer = styled(ContentWrapper)`
+  && {
+    width: 100%;
+    height: 100%;
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    align-items: center;
+  }
+`;
+
+/* use this to feed with mockData */
+const productsOnSale = mockData.items.filter(item => item.isOnSale);
+const data = { products: productsOnSale };
 
 const DealsPage = () => {
-  return <div>Deals</div>;
+  // const [data, setData] = useState({ products: [], isFetching: false });
```

Comment:
you are on the right track here! It might make things easier to have `isFetching` as a separate `useState` hook.